types.d.ts

```diff
@@ -1,35 +1,46 @@
-import { ReactElement } from 'react';
+import { Component, ReactElement } from 'react';
 import { StyleProp, ViewStyle, TextStyle } from 'react-native';
 
-declare module 'rn-swipe-button' {
-    class RNSwipeButton {
-        containerStyle?: StyleProp<ViewStyle>;
-        disabled?: boolean;
-        disabledRailBackgroundColor?: string;
-        disabledThumbIconBackgroundColor?: string;
-        disabledThumbIconBorderColor?: string;
-        enableRightToLeftSwipe?: boolean;
-        height?: number | string;
-        onSwipeFail?: () => void;
-        onSwipeStart?: () => void;
-        onSwipeSuccess?: () => void;
-        railBackgroundColor?: string;
-        railBorderColor?: string;
-        railFillBackgroundColor?: string;
-        railFillBorderColor?: string;
-        resetAfterSuccessAnimDuration?: number;
-        screenReaderEnabled?: boolean;
-        shouldResetAfterSuccess?: boolean;
-        swipeSuccessThreshold?: number; // Ex: 70. Swipping 70% will be considered as successful swipe
-        thumbIconBackgroundColor?: string;
-        thumbIconBorderColor?: string;
-        thumbIconComponent?: ReactElement;
-        thumbIconImageSource?: string | number;
-        thumbIconStyles?: StyleProp<ViewStyle>;
-        title?: string;
-        titleColor?: string;
-        titleFontSize?: number;
-        titleStyles?: StyleProp<TextStyle>;
-        width?: string | number;
-    }
-}
\ No newline at end of file
+interface Props {
+    containerStyle?: StyleProp<ViewStyle>;
+    disabled?: boolean;
+    disabledRailBackgroundColor?: string;
+    disabledThumbIconBackgroundColor?: string;
+    disabledThumbIconBorderColor?: string;
+    enableRightToLeftSwipe?: boolean;
+    height?: number | string;
+    onSwipeFail?: () => void;
+    onSwipeStart?: () => void;
+    onSwipeSuccess?: () => void;
+    railBackgroundColor?: string;
+    railBorderColor?: string;
+    railFillBackgroundColor?: string;
+    railFillBorderColor?: string;
+    resetAfterSuccessAnimDuration?: number;
+    screenReaderEnabled?: boolean;
+    shouldResetAfterSuccess?: boolean;
+    swipeSuccessThreshold?: number; // Ex: 70. Swipping 70% will be considered as successful swipe
+    thumbIconBackgroundColor?: string;
+    thumbIconBorderColor?: string;
+    thumbIconComponent?: ReactElement;
```

Comment:
I did a test for this pull. I think this is should be () => ReactElement

Reply:
This should be latest prop type `PropTypes.elementType`. @arelstone I'm approving this PR, its your call, you want to do it part of this PR or not. Thanks